lib/logstash/outputs/s3/uploader.rb

```diff
@@ -44,15 +46,22 @@ def upload(file, options = {})
             #
             # Thread might be stuck here, but I think its better than losing anything
             # its either a transient errors or something bad really happened.
-            logger.error("Uploading failed, retrying.", :exception => e.class, :message => e.message, :path => file.path, :backtrace => e.backtrace)
-            sleep TIME_BEFORE_RETRYING_SECONDS
-            retry
+            if tries < @retry_count
+              tries += 1
+              logger.warn("Uploading failed, retrying (#{tries}).", :exception => e.class, :message => e.message, :path => file.path, :backtrace => e.backtrace)
```

Comment:
We default logging to info, so by default it will still show as before, but I don't think it's a big concern as logging levels can be tuned per plugin

Btw, Maybe adding the extra information about total number of retries is also useful. for example:
```suggestion
              logger.warn("Uploading failed, retrying (##{tries} of #{@retry_count})", :exception => e.class, :message => e.message, :path => file.path, :backtrace => e.backtrace)
```

to produce:
```ruby
logger.warn("Uploading failed, retrying (#1 of 4)", :exception => "...", :message => "..", :path => "..", :backtrace => ["..."])
logger.warn("Uploading failed, retrying (#2 of 4)", :exception => "...", :message => "..", :path => "..", :backtrace => ["..."])

```

Reply:
cool, thanks. we could at least tell users to raise the plugin's level to error :wine_glass: